bin/varnishd/mgt/mgt_child.c

```diff
@@ -588,12 +599,26 @@ MGT_Child_Cli_Fail(void)
 		return;
 	if (child_pid < 0)
 		return;
-	MGT_complain(C_ERR, "Child (%jd) not responding to CLI, killing it.",
-	    (intmax_t)child_pid);
-	if (MGT_FEATURE(FEATURE_NO_COREDUMP))
-		(void)kill(child_pid, SIGKILL);
-	else
-		(void)kill(child_pid, SIGQUIT);
+	if (kill_child() == 0)
+		MGT_complain(C_ERR, "Child (%jd) not responding to CLI,"
+		    " killing it.", (intmax_t)child_pid);
+	else {
+		switch(errno) {
+		case EPERM:
+			MGT_complain(C_ERR, "Child (%jd) not responding to CLI."
+			    " Tried to kill it, but did not have the rights.",
+			    (intmax_t)child_pid);
+			/* Here we detected a bug in the jail implementation. */
+			break;
+		case ESRCH:
+			MGT_complain(C_ERR, "Child (%jd) not responding to CLI."
+			    " Tried to kill it, but PID did not exist.",
+			    (intmax_t)child_pid);
+			break;
+		default:
+			WRONG("Got wrong error from kill()");
+		}
+	}
```

Comment:
This is an old commit - I was asked to remove verbosity and in the latest version this code is different.